sample_workloads/nccltest/docker/scripts/container_entry.sh

```diff
@@ -0,0 +1,159 @@
+#!/bin/bash
+
+# Container entry script. Runs NCCL-level benchmarks and report results.
+
+set -u
+
+# Job parameters.
+: "${JOB_TIMESTAMP:?Must set JOB_TIMESTAMP}"
+: "${JOB_NAME:?Must set JOB_NAME}"
+: "${MASTER_ADDR:?Must set MASTER_ADDR}"
+: "${NNODES:?Must set NNODES}"
+: "${NODE_RANK:?Must set NODE_RANK}"
+
+# Benchmark parameters.
+: "${BENCHMARKS_CSV:?Must set BENCHMARKS_CSV}"
+: "${MASKS_CSV:?Must set MASKS_CSV}"
+: "${MSG_SIZE_BEGIN:?Must set MSG_SIZE_BEGIN}"
+: "${MSG_SIZE_END:?Must set MSG_SIZE_END}"
+: "${GPUS_PER_NODE:?Must set GPUS_PER_NODE}"
+: "${WARMUP_ITERS:?Must set WARMUP_ITERS}"
+: "${RUN_ITERS:?Must set RUN_ITERS}"
+: "${N_RUNS:?Must set N_RUNS}"
+
+# Unreserved cores for taskset call. This is a CSV of ranges for cores unused
+# by TCPX.
+: "${UNRESERVED_CORES:?Must set UNRESERVED_CORES}"
+
+# Telemetry.
+: "${GPU_TELEMETRY:?Must set GPU_TELEMETRY}"
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+# If input is malformed, error straight away.
+IFS=',' read -r -a BENCHMARKS <<< "$BENCHMARKS_CSV"
+IFS=',' read -r -a MASKS <<< "$MASKS_CSV"
+
+if [[ "${#BENCHMARKS[@]}" -ne "${#MASKS[@]}" ]]; then
+  echo "Mismatching length of BENCHMARKS_CSV and MASKS_CSV, aborting..."
+  exit 1
+fi
+
+for BENCHMARK in ${BENCHMARKS[@]}; do
+  if [[ "$BENCHMARK" != "all_gather_perf"     && "$BENCHMARK" != "all_reduce_perf" && \
+        "$BENCHMARK" != "reduce_scatter_perf" && "$BENCHMARK" != "broadcast_perf" && \
+        "$BENCHMARK" != "reduce_perf"         && "$BENCHMARK" != "sendrecv_perf" && \
+        "$BENCHMARK" != "scatter_perf"        && "$BENCHMARK" != "gather_perf" && \
+        "$BENCHMARK" != "alltoall_perf"       && "$BENCHMARK" != "hypercube_perf" ]]; then
+    echo "${BENCHMARK} is not a legal benchmark, aborting..."
+    exit 2
+  fi
+done
+
+for MASK in ${MASKS[@]}; do
```

Comment:
We don't necessarily have to have this limit for this public version; up to you whether you want to keep it. If you want to change it, also change line 15 of values.yaml and lines 110 and 111 in README.